rules/go_rules.build_defs

```diff
@@ -943,13 +943,14 @@ def go_mod_download(name:str, module:str, version:str, test_only:bool=False, vis
         tools = [CONFIG.GO_TOOL],
         building_description = 'Fetching...',
         cmd = ' && '.join(cmds),
-        requires = ['go'],
+        requires = ['go_src'],
         test_only = test_only,
         labels = labels + ['link:plz-out/go'],
         hashes = hashes,
         sandbox = False,
         licences = licences,
         visibility = visibility,
+        deps = deps,
```

Comment:
This allows two download rules to output to the same path as they become dependent on each-other avoiding race conditions